src/notebooks/R Mia Examples/Comparative-Metagenomics.qmd

```diff
@@ -0,0 +1,852 @@
+```
+
+# Fetching data and preprocessing
+In this example we are going to fetch MGnify analysis results and metadata for a dataset on the human gut microbiome viewed across age and geography -  ([MGYS00005154](https://www.ebi.ac.uk/metagenomics/studies/MGYS00005154)).
+
+## Setting the study accession
+1. First, set the accession number of the study we're using to the `mgnify_study_accession` variable.
+
+```{r}
+#| output: false
+
+source("./lib/variable_utils.r")
+
+mgnify_study_accession <- get_variable_from_link_or_input('MGYS', 'Study Accession', 
+                                                          'MGYS00005154')
+
+# You can also just directly set the accession variable in code, like this:
+# mgnify_study_accession <- "MGYS00005292"
+```
+
+2. Then, construct a MgnifyClient object to access the database.
+
+```{r}
+#| output: false
+mg <- MgnifyClient(usecache = TRUE, cache_dir = '/home/jovyan/.mgnify_cache')
+```
+
+3. Fetch a list of the Analyses for the Study.
+
+```{r}
+#| output: false
+analyses_accessions <- searchAnalysis(mg, "studies", mgnify_study_accession)
+analyses_accessions
+```
+
+## Download metadata
+1. First, download the metadata for the first 20 Analyses and put it into a dataframe.
+
+```{r}
+#| output: false
+analyses_metadata_df <- getMetadata(mg, head(analyses_accessions,
+                                                              30))
+```
+
+2. The table could be big, so let's look at a sample of it (`head`)
+
+```{r}
+#| output: false
+t(head(analyses_metadata_df))
+```
+
+## Convert to [TreeSummarizedExperiment object using mia](https://microbiome.github.io/)
+
+```{r}
+#| output: false
+tse <- getResult(mg, analyses_metadata_df$analysis_accession)
+```
+
+# Normalization, alpha diversity indices and taxonomic profiles visualization
+## Cleaning the OTUs matrix
+1. First, identify the total counts per sample and add them to the colData of our `tse` object.
+
+```{r}
+tse <- addPerCellQC(tse)
+```
+
+2. We then plot a histogram to identify outliers, so we can remove samples with low coverage.
+
+```{r}
+library_sie_histogram <- ggplot(colData(tse)) +
+  geom_histogram(aes(x = sum), color = "black", 
+                 fill = "gray", bins = 50) +
+  labs(x = "Library size", y = "Frequency (n)") + 
+  scale_x_log10(breaks = scales::trans_breaks("log10", function(x) 10^x), 
+                labels = scales::trans_format("log10", scales::math_format(10^.x))) +
+  theme_bw() +
+  # Remove the grid
+  theme(panel.grid.major = element_blank(), 
+        panel.grid.minor = element_blank(),
+        panel.border = element_blank(),
+        panel.background = element_blank(),
+        # Add y-axis line
+        axis.line = element_line(colour = "black"))
+```
+
+We can see that samples with number of reads below 10^1.6 seem to be outliers. Let’s filter out the outliers and plot a new histogram.
+
+```{r}
+filtered_data <- colData(tse)[colData(tse)$sum > 10^1.6, ]
```

Comment:
For plotColData I think you could just do the subsetting for the TreeSE object?

Like


`tse.filtered <- tse[, tse$sum > 1e1.6]`

Reply:
Then colData(tse.filtered) is the same than your current filtered_data